machine/v1/types_nutanixprovider.go

```diff
@@ -117,35 +128,6 @@ const (
 	NutanixSecureBoot NutanixBootType = "SecureBoot"
 )
 
-// NutanixIdentifierType is an enumeration of different resource identifier types.
-type NutanixIdentifierType string
-
-const (
-	// NutanixIdentifierUUID is a resource identifier identifying the object by UUID.
-	NutanixIdentifierUUID NutanixIdentifierType = "uuid"
-
-	// NutanixIdentifierName is a resource identifier identifying the object by Name.
-	NutanixIdentifierName NutanixIdentifierType = "name"
-)
-
-// NutanixResourceIdentifier holds the identity of a Nutanix PC resource (cluster, image, subnet, etc.)
-// +union
-type NutanixResourceIdentifier struct {
-	// Type is the identifier type to use for this resource.
-	// +unionDiscriminator
-	// +kubebuilder:validation:Required
-	// +kubebuilder:validation:Enum:=uuid;name
-	Type NutanixIdentifierType `json:"type"`
-
-	// uuid is the UUID of the resource in the PC.
-	// +optional
-	UUID *string `json:"uuid,omitempty"`
-
-	// name is the resource name in the PC
-	// +optional
-	Name *string `json:"name,omitempty"`
-}
-
```

Comment:
These should stay here. These types have not gone through API review, where the configv1 types will. As I'm able to request changes in the new additions, do not change or re-use these old ones, or try to re-use the old ones here. The old API must remain the same, or be changed in a compatible way

Reply:
Do you mean to keep the old NutanixIdentifierType and NutanixResourceIdentifier in the machine/v1 package, and to define new types of these in the config/v1 for the NutanixFailureDomain use?

Reply:
Changed as requested.